src/IK/src/InverseKinematics.cpp

```diff
@@ -69,32 +157,38 @@ double HumanIK::getDt() const
     return std::chrono::duration_cast<std::chrono::duration<double>>(m_dtIntegration).count();
 }
 
-bool HumanIK::setDoFsNumber(const int nrDoFs)
-{
-    m_nrDoFs = nrDoFs;
-    m_jointPositions.resize(m_nrDoFs);
-    m_jointVelocities.resize(m_nrDoFs);
-
-    return true;
-}
-
 int HumanIK::getDoFsNumber() const
 {
     return m_nrDoFs;
 }
 
-bool HumanIK::setInitialJointPositions(const Eigen::Ref<const Eigen::VectorXd> qInitial)
+bool HumanIK::setNodeSetPoint(const int node,
+                              const manif::SO3d& I_R_IMU,
+                              const manif::SO3Tangentd& I_omega_IMU)
 {
-    m_jointPositions = qInitial;
-
-    return true;
+    bool ok;
+    if (m_OrientationTasks.find(node) == m_OrientationTasks.end())
+    {
+        BiomechanicalAnalysis::log()->error("[HumanIK::setNodeSetPoint] Invalid node number.");
+        return false;
+    }
+    I_R_link = m_OrientationTasks[node].calibrationMatrix * I_R_IMU
+               * m_OrientationTasks[node].IMU_R_link;
+    ok = m_OrientationTasks[node].task->setSetPoint(I_R_link, I_omega_IMU);
+    return ok;
 }
 
-bool HumanIK::setLink1OrientationAndAngVel(const manif::SO3d &link1Orientation,
-                                           const manif::SO3Tangentd &link1AngularVelocity)
+bool HumanIK::TPoseCalibrationNode(const int node, const manif::SO3d& I_R_IMU)
 {
-    m_link1OrientationTask->setSetPoint(link1Orientation, link1AngularVelocity);
-
+    if (m_OrientationTasks.find(node) == m_OrientationTasks.end())
+    {
+        BiomechanicalAnalysis::log()->error("[HumanIK::setNodeSetPoint] Invalid node number.");
+        return false;
+    }
+    m_OrientationTasks[node].calibrationMatrix
+        = calib_R_link * (I_R_IMU * m_OrientationTasks[node].IMU_R_link).inverse();
+    std::cout << "calibration matrix of node " << node << ": "
+              << m_OrientationTasks[node].calibrationMatrix << std::endl;
```

Comment:
I don't know if this is needed. If yes you can use `log()->`

Reply:
Actually this was just for debugging purposes, now I deleted it.